File: frontend/src/types/competitor.ts
export interface Competitor {
  id: string
  name: string
  platform: string
  avg_rating: number | null
  review_count: number
  url: string | null
  created_at: string
}

export interface CompetitorListResponse {
  competitors: Competitor[]
  total: number
}

export interface LocationRanking {
  location_id: string
  avg_rating: number
  review_count: number
  sentiment_breakdown: Record<string, number>
  positive_percentage: number
  rank: number
}

export interface LeaderboardResponse {
  locations: LocationRanking[]
}

File: backend/app/schemas/competitor.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class CompetitorCreate(BaseModel):
    name: str
    platform: str
    avg_rating: float | None = None
    review_count: int = 0
    url: str | None = None


class CompetitorUpdate(BaseModel):
    name: str | None = None
    platform: str | None = None
    avg_rating: float | None = None
    review_count: int | None = None
    url: str | None = None


class CompetitorResponse(BaseModel):
    id: uuid.UUID
    name: str
    platform: str
    avg_rating: float | None
    review_count: int
    url: str | None
    created_at: datetime

    class Config:
        from_attributes = True


class CompetitorListResponse(BaseModel):
    competitors: list[CompetitorResponse]
    total: int
